### PR TITLE
GCS store fixes

### PR DESCRIPTION
**Pull request by fabxc**

### CODE REVIEW

File: pkg/store/gcs.go [hunk 1]
```diff
@@ -205,17 +213,105 @@ func (s *GCSStore) Info(context.Context, *storepb.InfoRequest) (*storepb.InfoRes
...
 	matchers, err := translateMatchers(req.Matchers)
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 	var g errgroup.Group
 	// TODO(fabxc): filter inspected blocks by labels.
 
+	res := &seriesResult{res: map[uint64]*seriesEntry{}}
+
 	for _, b := range s.blocks {
```
Comment (Bplotka) [inline pkg/store/gcs.go]: I think you need mutex here, right?

Comment (fabxc) [reply pkg/store/gcs.go]: Yes, added.

File: pkg/store/gcs.go [hunk 2]
```diff
@@ -65,6 +67,9 @@ func NewGCSStore(logger log.Logger, bucket *storage.BucketHandle, dir string) (*
 
 // Close the store.
 func (s *GCSStore) Close() (err error) {
+	s.mtx.RLock()
```
Comment (Bplotka) [inline pkg/store/gcs.go]: Can we switch that to just `s.mtx.Lock()` (RWMutext to just Mutex?) 
We don't use the RW logic

File: pkg/store/gcs.go [hunk 3]
```diff
@@ -205,17 +226,107 @@ func (s *GCSStore) Info(context.Context, *storepb.InfoRequest) (*storepb.InfoRes
 	return &storepb.InfoResponse{}, nil
 }
 
+type seriesResult struct {
+	mtx sync.RWMutex
```
Comment (Bplotka) [inline pkg/store/gcs.go]: What about moving to `mtx.Mutex` we don't use RW logic

File: pkg/store/gcs.go [hunk 4]
```diff
@@ -226,52 +337,44 @@ func (s *GCSStore) Series(ctx context.Context, req *storepb.SeriesRequest) (*sto
...
-				}
 				for _, meta := range chunks {
 					if meta.MaxTime < req.MinTime {
 						continue
 					}
 					if meta.MinTime > req.MaxTime {
 						break
 					}
-					c, err := b.chunks.Chunk(meta.Ref)
+					meta.Chunk, err = block.chunks.Chunk(meta.Ref)
```
Comment (Bplotka) [inline pkg/store/gcs.go]: Should we add retry logic here? in case of transient errors?

Comment (Bplotka) [reply pkg/store/gcs.go]: or just add TODO for it now

**Pull request merged**